daemon/containerd/image.go

```diff
@@ -48,11 +48,19 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 	err = i.walkImageManifests(ctx, desc, func(img *ImageManifest) error {
 		conf, err := img.Config(ctx)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to get config descriptor: %w", err)
```

Comment:
Should these new errors have a specific type? e.g., should this be a `NotFound` error?

Reply:
Oh, I guess the reverse (as we handle NotFound different?)

Reply:
Hmm, there's a very small chance this could fail with `NotFound` because this is only reading the config descriptor from the manifest content. The only way this could fail with `NotFound` is if the manifest is missing or malformed, but then it wouldn't be considered by the `walkImageManifests`.
In theory this could happen if the content was deleted between `walkImageManifests` picked it up and the handler func was called.

So I guess it doesn't hurt to handle `NotFound` here, but return `System` in other cases.